java-runtime-builder-app/src/main/java/com/google/cloud/runtimes/builder/buildsteps/RuntimeImageBuildStep.java

```diff
@@ -70,42 +73,42 @@ public void run(BuildContext buildContext) throws BuildStepException {
   private String getBaseRuntimeImage(BuildContext buildContext, Artifact artifact)
       throws BuildStepException {
     RuntimeConfig runtimeConfig = buildContext.getRuntimeConfig();
+    ArtifactType artifactType = artifact.getType();
 
-    // Check if the user has explicitly selected the compat runtime
-    if (buildContext.isCompatEnabled() && artifact.getType() != EXPLODED_WAR) {
-      throw new BuildStepException(String.format("App Engine APIs have been enabled. In order to "
-          + "use App Engine APIs, an exploded WAR artifact is required, but a %s artifact was "
-          + "found. See https://cloud.google.com/appengine/docs/flexible/java/upgrading for more "
-          + "detail.", artifact.getType()));
+    if (artifactType == COMPAT_EXPLODED_WAR) {
+      return compatImageName;
```

Comment:
There are still 3 return statements in this method, and we're missing the the log statement `Using base image '{}' for {} artifact` for compat. I think it will be much clearer if you just refactor the method to have a single if-elseif-else block followed by `logger.info`, followed by a `return`.

Reply:
Done